lib/controllers/root-controller.js

```diff
@@ -362,13 +367,15 @@ export default class RootController extends React.Component {
   }
 
   @autobind
-  async initializeRepo() {
+  async initializeRepo(initDialogPath) {
     if (this.props.activeWorkingDirectory) {
       await this.acceptInit(this.props.activeWorkingDirectory);
-      return;
+      return null;
     }
 
-    this.setState({initDialogActive: true});
+    return new Promise(resolve => {
+      this.setState({initDialogActive: true, initDialogPath, initDialogResolve: resolve});
+    });
```

Comment:
Storing the promise resolution callback in the state like this feels a bit weird to me. I have no cleaner ideas for how to structure this though :grin:

Is it possible for this `setState()` call to be triggered twice before the first resolution callback is triggered... ?